src/status_im2/contexts/chat/lightbox/zoomable_image/view.cljs

```diff
@@ -0,0 +1,313 @@
+(ns status-im2.contexts.chat.lightbox.zoomable-image.view
+  (:require
+   [react-native.core :as rn]
+   [react-native.gesture :as gesture]
+   [react-native.reanimated :as reanimated]
+   [reagent.core :as reagent]
+   [utils.re-frame :as rf]))
+
+;; Definitions
+(def min-scale 1)
+
+(def double-tap-scale 2)
+
+(def max-scale 5)
+
+(def init-offset 0)
+
+(def velocity-factor 0.25)
+
+(def default-duration 300)
+
+;; Some aliases for reanimated methods, as they are used 10s of times in this file
+(defn get-val
+  [animation]
+  (reanimated/get-shared-value animation))
+
+(defn set-val
+  [animation value]
+  (reanimated/set-shared-value animation value))
+
+(defn use-val
+  [value]
+  (reanimated/use-shared-value value))
+
+(defn timing
+  ([value]
+   (timing value default-duration))
+  ([value duration]
+   (reanimated/with-timing-duration value duration)))
+
+(defn set-val-decay
+  [animation value bounds]
+  (reanimated/animate-shared-value-with-decay animation value bounds))
+
+;; MATH
+(defn get-max-offset
+  [size screen-size scale]
+  (/ (/ (- (* size scale) screen-size) 2) scale))
+
+(defn get-scale-ratio
+  [new-scale saved-scale]
+  (/ (- (dec new-scale) (dec saved-scale)) new-scale))
+
+(defn get-current-center
+  [size scaled-size offset]
+  (- (+ (/ (- size scaled-size) 2) (/ scaled-size 2)) offset))
+
+;; 5 Gestures: tap, double-tap, pinch, pan-x, pan-y
+(defn tap-gesture
+  [on-tap]
+  (->
+    (.Tap gesture/gesture)
+    (.onStart #(on-tap))))
+
+(defn double-tap-gesture
+  [{:keys [scale pan-x pan-x-start pan-y pan-y-start]}
+   {:keys [width height screen-height]}
+   {:keys [y-threshold-scale]}
+   rescale]
+  (->
+    (.Tap gesture/gesture)
```

Comment:
so the idea of having react-native.* namespaces is to hide all interactions with js objects, so it would be cool to move tap and numbersOfTaps to gesture namespace, we already have onStart, but we need to make it more general and remove pan argument

Reply:
@flexsurfer fixed it.